src/mmw/apps/modeling/geoprocessing.py

```diff
@@ -123,169 +218,42 @@ def sjs_retrieve(job_id, retry=None):
-    function which takes a list of polygons or multipolygons as input,
-    and returns histograms of the NLCD x soil pairs that occur in each
-    shape.
-
-    This is the top-half of the function.
-    """
-    data = settings.GEOP['json']['nlcd_soil_census'].copy()
-    data['input']['polygon'] = polygons
-
-    return sjs_submit(data, retry)
-
-
-@statsd.timer(__name__ + '.histogram_finish')
-def histogram_finish(job_id, retry):
+def parse(sjs_result):
     """
-    This is the bottom-half of the function.
-    """
-    sjs_result = sjs_retrieve(job_id, retry)
+    Converts raw JSON results from Spark JobServer to dictionary of tuples
 
-    # Convert string "List(3,4)" to tuples (3,4) and
-    # Map NODATA soil cells cells to 3
-    # This was previously done within the SummaryJob: https://github.com/WikiWatershed/mmw-geoprocessing/blob/0d95dee35e729d9fd2f58fb9e73a69dcbe61df61/summary/src/main/scala/SummaryJob.scala#L106  # NOQA
-    # but since MapshedJob doesn't support remapping, we do it here.
-    nlcd_soil_count = {}
-    for key, count in sjs_result.iteritems():
-        (n, s) = make_tuple(key[4:])  # Convert "List(3,4)" to (3, 4)
-        s2 = s if s != settings.NODATA else 3  # Map NODATA soil cells to 3
-        nlcd_soil_count[(n, s2)] = count + nlcd_soil_count.get((n, s2), 0)
+    If the input is this:
 
-    # Convert to array for backwards compatibility
-    result = [(ns, count) for ns, count in nlcd_soil_count.items()]
+        {
+            'List(1,2)': 3,
+            'List(4,5)': 6
+        }
 
-    return [result]
+    The output will be:
 
+        {
+            (1, 2): 3,
+            (4, 5): 6
+        }
 
-def histogram_to_x(data, nucleus, update_rule, after_rule):
-    """
-    Transform a raw Geotrellis histogram into an object of the desired
-    form (dictated by the last three arguments).
+    :param sjs_result: Dictionary mapping strings like 'List(a,b,c)' to ints
+    :return: Dictionary mapping tuples of ints to ints
     """
-    retval = nucleus
-    total = 0
+    return {make_tuple(key[4:]): val for key, val in sjs_result.items()}
 
-    for ((nlcd, soil), count) in data:
-        total += count
-        update_rule(nlcd, soil, count, retval)
 
-    after_rule(total, retval)
-
-    return retval
-
-
-@statsd.timer(__name__ + '.data_to_censuses')
-def data_to_censuses(data):
-    return [data_to_census(subdata) for subdata in data]
-
-
-def data_to_census(data):
+def to_one_ring_multipolygon(area_of_interest):
     """
-    Turn raw data from Geotrellis into a census.
+    Given a multipolygon comprising just a single ring structured in a
+    five-dimensional array, remove one level of nesting and make the AOI's
+    coordinates a four-dimensional array. Otherwise, no op.
     """
-    def update_rule(nlcd, soil, count, census):
-        dist = census['distribution']
-        if nlcd in settings.NLCD_MAPPING and soil in settings.SOIL_MAPPING:
```

Comment:
Without this condition, getting a `KeyError` for values not in `NLCD_MAPPING` (the "0")
```
[2017-05-15 14:55:02,295: ERROR/Worker-1] Task 752ab6d5-96a0-4634-bff6-e6454d5279ea run from job 7097 raised exception: 0
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/app/apps/modeling/tasks.py", line 364, in nlcd_soil_census
    nlcd=settings.NLCD_MAPPING[n][0])
KeyError: 0
```

Reproducible on Schuykill HUC-08, though it may occasionally time out